From 5359440ac6e03f62c6fe315e58474a0c28e4f055 Mon Sep 17 00:00:00 2001
From: Ivan Chavero <ichavero@redhat.com>
Date: Thu, 14 May 2015 17:15:12 -0600
Subject: [PATCH] Add ability to enable rdo testing repo

The new rdo-release-kilo now adds the testing repo this patch
allows the user to enable the repo.

Change-Id: I2de49561ca287df42e084e3d6c70183f7a1e21f9
---
 docs/packstack.rst                 |  3 +++
 packstack/plugins/prescript_000.py | 19 ++++++++++++++++++-
 2 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/docs/packstack.rst b/docs/packstack.rst
index 9060016..cdefb70 100644
--- a/docs/packstack.rst
+++ b/docs/packstack.rst
@@ -134,6 +134,9 @@
 **CONFIG_REPO**
     Comma-separated list of URLs for any additional yum repositories, to use for installation.
 
+**CONFIG_ENABLE_RDO_TESTING**
+   Specify 'y' to enable the RDO testing repository. ['y', 'n']
+
 RHEL config
 -----------
 
diff --git a/packstack/plugins/prescript_000.py b/packstack/plugins/prescript_000.py
index 62d8fa2..ee4aa38 100644
--- a/packstack/plugins/prescript_000.py
+++ b/packstack/plugins/prescript_000.py
@@ -539,7 +539,19 @@
              "CONF_NAME": "CONFIG_REPO",
              "USE_DEFAULT": False,
              "NEED_CONFIRM": False,
-             "CONDITION": False}
+             "CONDITION": False},
+
+            {"CMD_OPTION": "enable-rdo-testing",
+             "PROMPT": "To enable rdo testing enter \"y\"",
+             "OPTION_LIST": ["y", "n"],
+             "VALIDATORS": [validators.validate_options],
+             "DEFAULT_VALUE": "n",
+             "MASK_INPUT": False,
+             "LOOSE_VALIDATION": True,
+             "CONF_NAME": "CONFIG_ENABLE_RDO_TESTING",
+             "USE_DEFAULT": False,
+             "NEED_CONFIRM": False,
+             "CONDITION": False},
         ],
 
         "RHEL": [
@@ -1105,6 +1117,11 @@
     reponame = 'openstack-%s' % version
     server.clear()
     server.append('yum-config-manager --enable %(reponame)s' % locals())
+
+    if config['CONFIG_ENABLE_RDO_TESTING'] == 'y':
+        server.append('yum-config-manager --disable %(reponame)s' % locals())
+        server.append('yum-config-manager --enable %(reponame)s-testing' % locals())
+
     # yum-config-manager returns 0 always, but returns current setup
     # if succeeds
     rc, out = server.execute()
